vortex-duckdb-ext/src/scan.rs

```diff
@@ -140,8 +173,19 @@ impl TableFunction for VortexTableFunction {
         Ok(())
     }
 
-    fn init_global(_init: &TableInitInput<Self>) -> VortexResult<Self::GlobalState> {
-        Ok(VortexGlobalData {})
+    fn init_global(init: &TableInitInput<Self>) -> VortexResult<Self::GlobalState> {
+        let bind_data = init.bind_data();
+        let file_paths = SegQueue::new();
+
+        // Skip the first file path, as the file is opened during bind.
+        for path in bind_data.file_paths.iter().skip(1) {
+            file_paths.push(path.clone());
```

Comment:
can you not Extend?

Reply:
`extend` doesn't exist on the lockfree `SegQueue`.